docs/pages/database-access/guides/postgres-cloudsql.mdx

```diff
@@ -0,0 +1,336 @@
+
+Save the file, your Teleport Database Service will need it to be able to generate
+IAM auth tokens.
+
+## Gather Cloud SQL Instance Information
+
+To connect a Cloud SQL database to Teleport, you'll need to gather a few pieces
+of information about the instance.
+
+- GCP Project ID.
+
+You can normally see it in the organization view at the top of the GCP dashboard.
+
+- Cloud SQL instance ID.
+
+The instance ID is the name of your Cloud SQL instance shown at the top of the
+Overview page:
+
+![Instance ID](../../../img/database-access/guides/cloudsql/instance-id.png)
+
+- Cloud SQL instance endpoint.
+
+You will use the instance's public IP address to connect to it. It can be viewed
+on the "Connect to this instance" panel on the Overview page:
+
+![Instance Public IP](../../../img/database-access/guides/cloudsql/instance-public-ip.png)
+
+- Cloud SQL instance root certificate.
+
+The instance's root certificate is required so Teleport can validate the
+certificate presented by the database instance. You can download `server-ca.pem`
+file from the Connections tab under Security section:
+
+![Instance Root Certificate](../../../img/database-access/guides/cloudsql/instance-root-ca.png)
+
+## Configure Teleport
+
+Cloud SQL support for Teleport Database Access is available starting from
+`6.1` release.
+
+Download the appropriate version of Teleport for your platform from
+our [downloads page](https://goteleport.com/teleport/download).
+
+Follow the installation [instructions](../../installation.mdx).
+
+### Start Auth/Proxy Service
+
+Create a configuration file for a Teleport service that will be running
+auth and proxy servers:
+
+```yaml
+teleport:
+  data_dir: /var/lib/teleport
+  nodename: test
+auth_service:
+  enabled: "yes"
+proxy_service:
+  enabled: "yes"
+  # Set public address proxy will be reachable at.
+  public_addr: teleport.example.com:3080
+  # The PostgreSQL proxy listens on the regular web proxy port.
+  web_listen_addr: 0.0.0.0:3080
+ssh_service:
+  enabled: "no"
+```
+
+Start the service:
+
+```sh
+$ teleport start --config=/path/to/teleport.yaml
+```
+
+Generate a short-lived join token for the database service and save it for
+example in `/tmp/token`:
+
+```sh
+$ tctl tokens add \
+    --type=db \
+    --db-name=test \
+    --db-protocol=postgres \
+    --db-uri=35.1.2.3:5432
+```
+
+### Create Role and User
+
+Create a role that will allow a user to connect to any database using any
+database account:
+
+```sh
+$ tctl --config=/path/to/teleport.yaml create <<EOF
+kind: role
+version: v3
+metadata:
+  name: db
+spec:
+  allow:
+    db_labels:
+      '*': '*'
+    db_names:
+    - '*'
+    db_users:
+    - '*'
+EOF
+```
+
+Create a user and assign them the `db` role we've just created:
+
+```sh
+$ tctl --config=/path/to/teleport.yaml users add --roles=editor,access,db testuser
+```
+
+### Start Database Service with Config File
```

Comment:
@r0mant would be cool if you had a `teleport configure --preset=db-iam` one liner instead of this config and then referenced the proper config file at the end

Reply:
This is a good idea. I'll do it for 6.2 if I have time and update db quickstarts, created a ticket for now: https://github.com/gravitational/teleport/issues/6368.